docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,319 @@
+Continue: 'continue';
+Contract: 'contract';
+Delete: 'delete';
+Do: 'do';
+Else: 'else';
+Emit: 'emit';
+Enum: 'enum';
+Event: 'event';
+External: 'external';
+Fallback: 'fallback';
+False: 'false';
+Fixed: 'fixed' | ('fixed' [0-9]+ 'x' [0-9]+);
+From: 'from';
+/**
+ * Bytes types of fixed length.
+ * byte is an alias of bytes1.
+ */
+FixedBytes:
+	'byte' | 'bytes1' | 'bytes2' | 'bytes3' | 'bytes4' | 'bytes5' | 'bytes6' | 'bytes7' | 'bytes8' |
+	'bytes9' | 'bytes10' | 'bytes11' | 'bytes12' | 'bytes13' | 'bytes14' | 'bytes15' | 'bytes16' |
+	'bytes17' | 'bytes18' | 'bytes19' | 'bytes20' | 'bytes21' | 'bytes22' | 'bytes23' | 'bytes24' |
+	'bytes25' | 'bytes26' | 'bytes27' | 'bytes28' | 'bytes29' | 'bytes30' | 'bytes31' | 'bytes32';
+For: 'for';
+Function: 'function';
+Hex: 'hex';
+If: 'if';
+Immutable: 'immutable';
+Import: 'import';
+Indexed: 'indexed';
+Interface: 'interface';
+Internal: 'internal';
+Is: 'is';
+Library: 'library';
+Mapping: 'mapping';
+Memory: 'memory';
+Modifier: 'modifier';
+New: 'new';
+/**
+ * Unit denomination for numbers.
+ */
+NumberUnit: 'wei' | 'gwei' | 'ether' | 'seconds' | 'minutes' | 'hours' | 'days' | 'weeks' | 'years';
+Override: 'override';
+Payable: 'payable';
+Private: 'private';
+Public: 'public';
+Pure: 'pure';
+Receive: 'receive';
+Return: 'return';
+Returns: 'returns';
+/**
+ * Sized signed integer types.
+ * int is an alias of int256.
+ */
+SignedIntegerType:
+	'int' | 'int8' | 'int16' | 'int24' | 'int32' | 'int40' | 'int48' | 'int56' | 'int64' |
+	'int72' | 'int80' | 'int88' | 'int96' | 'int104' | 'int112' | 'int120' | 'int128' |
+	'int136' | 'int144' | 'int152' | 'int160' | 'int168' | 'int176' | 'int184' | 'int192' |
+	'int200' | 'int208' | 'int216' | 'int224' | 'int232' | 'int240' | 'int248' | 'int256';
+Storage: 'storage';
+String: 'string';
+Struct: 'struct';
+True: 'true';
+Try: 'try';
+Type: 'type';
+Ufixed: 'ufixed' | ('ufixed' [0-9]+ 'x' [0-9]+);
+/**
+ * Sized unsigned integer types.
+ * uint is an alias of uint256.
+ */
+UnsignedIntegerType:
+	'uint' | 'uint8' | 'uint16' | 'uint24' | 'uint32' | 'uint40' | 'uint48' | 'uint56' | 'uint64' |
+	'uint72' | 'uint80' | 'uint88' | 'uint96' | 'uint104' | 'uint112' | 'uint120' | 'uint128' |
+	'uint136' | 'uint144' | 'uint152' | 'uint160' | 'uint168' | 'uint176' | 'uint184' | 'uint192' |
+	'uint200' | 'uint208' | 'uint216' | 'uint224' | 'uint232' | 'uint240' | 'uint248' | 'uint256';
+Using: 'using';
+View: 'view';
+Virtual: 'virtual';
+While: 'while';
+
+LParen: '(';
+RParen: ')';
+LBrack: '[';
+RBrack: ']';
+LBrace: '{';
+RBrace: '}';
+Colon: ':';
+Semicolon: ';';
+Period: '.';
+Conditional: '?';
+Arrow: '=>';
+
+Assign: '=';
+AssignBitOr: '|=';
+AssignBitXor: '^=';
+AssignBitAnd: '&=';
+AssignShl: '<<=';
+AssignSar: '>>=';
+AssignShr: '>>>=';
+AssignAdd: '+=';
+AssignSub: '-=';
+AssignMul: '*=';
+AssignDiv: '/=';
+AssignMod: '%=';
+
+Comma: ',';
+Or: '||';
+And: '&&';
+BitOr: '|';
+BitXor: '^';
+BitAnd: '&';
+Shl: '<<';
+Sar: '>>';
+Shr: '>>>';
```

Comment:
Since this list closely mirrors that of `Token.h`, why not keep these three uppercase only or why not change them in `Token.h` too?

Reply:
In antlr upper- and lower-case has special semantics (lexer rules start upper case, grammar rules start lower case). While I think all-upper-case lexer rules would work here, I'd say it's better to just stick to the convention?

Reply:
Any reason not to change them in `Token.h` (adding it as a "reduce technical debt issue")

Reply:
Any reason **to** change them in ``Token.h``? I'll not do anything like that in this PR in any case.